### ISSUE TITLE
Login with service principal

### ISSUE THREAD

**Issue by Fredrik189100** [yangl900/armclient-go#7]
Hi,
The original Armclient have the option to login using a service principal, either with password or certificate (or both). https://github.com/projectKudu/ARMClient/wiki/Login-and-Acquire-Tokens

It does not seems like this is possible with the GO-implementation. This feature would be very helpful for running the armclient inside scripts.

**Comment by yangl900** [comment]
Indeed! Will find some time to add that. Contributions always welcome!
